src/Reflectivity/Context.extension.st

```diff
@@ -0,0 +1,97 @@
+	currentSymbolicBytecodeIndex := 1.
+	currentBytecode := currentSymbolicBytecodes at:
+		                   currentSymbolicBytecodeIndex.
+	newSymbolicBytecodeIndex := 1.
+	newBytecode := newSymbolicBytecodes at: newSymbolicBytecodeIndex.
+	[ "While all bytecodes from current method have not been found in new method:"
+	currentSymbolicBytecodeIndex < currentSymbolicBytecodes size "=" ]
+		whileTrue: [ "If the two bytecodes are the same, then no offset need to be added, else it means a bytecode has been added in the new method and the offset should be increased by the size of this bytecode:"
+			currentBytecode := currentSymbolicBytecodes at:
+				                   currentSymbolicBytecodeIndex.
+			newBytecode := newSymbolicBytecodes at: newSymbolicBytecodeIndex.
+			(self
+				 currentBytecode: currentBytecode
+				 equalsToNewBytecode: newBytecode
+				 withCurrents: self compiledCode symbolicBytecodes
+				 withNews: aCompiledMethod symbolicBytecodes)
+				ifTrue: [
+				currentSymbolicBytecodeIndex := currentSymbolicBytecodeIndex + 1 ]
+				ifFalse: [ offset := offset + newBytecode bytes size ].
+			newSymbolicBytecodeIndex := newSymbolicBytecodeIndex + 1 ].
+	^ aCompiledMethod initialPC + offset
+]
+
+{ #category : '*Reflectivity' }
+Context >> currentBytecode: currentBytecode equalsToNewBytecode: newBytecode withCurrents: currentSymbolicBytecodes withNews: newSymbolicBytecodes [
+
+	| result |
+	result := currentBytecode description = newBytecode description.
+	^ result
+		  ifTrue: [ true ]
+		  ifFalse: [
+			  ((currentBytecode description beginsWith: 'jump') and: [
+				   newBytecode description beginsWith: 'jump' ])
+				  ifFalse: [ false ]
+				  ifTrue: [
+					  | currentSplitDescription newSplitDescription |
+					  currentSplitDescription := $: split:
+						                             currentBytecode description.
+					  newSplitDescription := $: split: newBytecode description.
+					  (currentSplitDescription size = 2 and: [
+						   newSplitDescription size = 2 and: [
+							   currentSplitDescription first = newSplitDescription first ] ])
+						  ifFalse: [ false ]
+						  ifTrue: [
+							  | currentJumpPc newJumpPc currentJumpBytecode newJumpBytecode |
+							  currentJumpPc := (currentSplitDescription at: 2) asInteger.
+							  newJumpPc := (newSplitDescription at: 2) asInteger.
+							  currentJumpBytecode := currentSymbolicBytecodes
+								                         detect: [ :csb |
+								                         csb offset = currentJumpPc ]
+								                         ifNone: [ ^ false ].
+							  newJumpBytecode := newSymbolicBytecodes
+								                     detect: [ :nsb | nsb offset = newJumpPc ]
+								                     ifNone: [ ^ false ].
+							  self
+								  currentBytecode: currentJumpBytecode
+								  equalsToNewBytecode: newJumpBytecode
+								  withCurrents: currentSymbolicBytecodes
+								  withNews: newSymbolicBytecodes ] ] ]
```

Comment:
Maybe there is, I haven't found anything better and I would be happy to find something better.

What I know is that we cannot use bytes because some bytecodes that are identical in both methods can have different bytes

Reply:
The SymbolicBytecode was made just for the inspector, that is why there is just one class with a string inside.

I was always thinking of turning it into a real bytecode class hierachy, with all the state (like jump targets) being state of these objects. Could that simplify the code?

Reply:
This would simplify the code, a lot

Reply:
This code is so big just because I need to parse a description to know if two bytecodes are jumps and if the bytecodes they jump to are the same, from the PCs I parsed in the descriptions